src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_kds.c

```diff
@@ -409,7 +440,8 @@ static int xocl_command_ioctl(struct xocl_dev *xdev, void *data,
 	}
 
 	xobj = to_xocl_bo(obj);
-	if (!xocl_bo_execbuf(xobj)) {
+	if (!is_valid_execbuf(xdev, xobj)) {
```

Comment:
xocl is checking the BO buffer here which is accessible by user space code. malicious user code can still mess it up during the check or right after the check and crash the system. not sure how this fix would work?

Reply:
You are right. I should check execbuf after copy it to kernel space.